tests/testthat/test-xgboost-develop.R

```diff
@@ -0,0 +1,76 @@
+context("Checking xgboost development")
```

Comment:
I was a little off in my testing.  I tested that the tests built, which they do. But after running `devtools::test()` as Yannick did, I got the same errors.  It seems that some of the wording of the tests is the problem like in Failure 9, we test whether or not the output says "3 rows were inserted into the SQL Server HCRWriteData1 table" but the actual output says "3 rows were inserted into the SQLiteHCRWriteData1 table."

So the new tests for XGBoost make sense and they are built well, but during testing is where something is giving problems.